### DISCUSSION TITLE
PyTorch-adapt Multi-label classifications

### DISCUSSION THREAD

**Discussion by AlexandrByzov** [KevinMusgrave/pytorch-adapt#100] [Q&A] [answered]
Hi Kevin,

I have a multi-label classification problem, and I wanted to use PyTorch-adapt. I wrote a code for four hooks, but the model weights do not change. Is there something I need to change in hooks for multi-label classifications to work?

Best regards,
Alex

**Comment by KevinMusgrave** [comment, marked as answer]
You need to change `CLossHook`'s loss function to `torch.nn.BCEWithLogitsLoss`. The hook you're using should allow you to pass in a custom `c_hook` or something like that. I can help more with this if you tell me which hook you're using.

An alternative is to "monkey patch" the init function of `CLossHook`:

```python
import torch
from pytorch_adapt.hooks import CLossHook

def init_modifier(method):
    def modify(self, *args, **kwargs):
        method(self, *args, **kwargs)
        self.loss_fn = torch.nn.BCEWithLogitsLoss()
    return modify

CLossHook.__init__ = init_modifier(CLossHook.__init__)
```

Then every hook that uses `CLossHook` will now be using `torch.nn.BCEWithLogitsLoss`. 

You might get a type error during training with `torch.nn.BCEWithLogitsLoss`, in which case you can use this class instead (it just converts the input labels to floating point):
```python
from pytorch_adapt.hooks.classification import BCEWithLogitsConvertLabels
```

**Reply by AlexandrByzov** [reply]
Very cool, thank you very much!!!
I do experimental research on domain adaptation techniques for ECG data, so I wanted to check most of the hooks)
